bloom-housing/bloom
# fix: can remove application pick up and drop off addresses

# Pull Request Template

## Issue

Addresses #1913


## Description

If you saved a draft listing with application pick up and/or drop off as the leasing agent address, you could then never toggle it off. The issue was that we are stripping every null value from the data object we send to add a listing because the backend was not validating null values and empty strings in a way that was expected.

When we were first sending an applicationPickUpAddressType as "leasingAgent" and then just removing it from the data object and sending nothing on "no" the backend wasn't updating the field because we weren't sending it as null.

If the backend rejected empty strings for string fields and took the absence of a field as changing it to a null value we wouldn't have to do this.

## Type of change


## How Can This Be Tested/Reviewed?

If you saved a draft listing with application pick up and/or drop off as the leasing agent address, you could then never toggle it off. Ensure you can toggle it off.

## Checklist:
